eZ/Publish/Core/Persistence/Legacy/Content/Gateway/DoctrineDatabase.php

```diff
@@ -1523,7 +1530,7 @@ public function setName($contentId, $version, $name, $language)
             $q->bindValue($version, null, \PDO::PARAM_INT)
         )->set(
             $this->dbHandler->quoteColumn('language_id'),
-            $q->bindValue($language->id, null, \PDO::PARAM_INT)
+            $q->bindValue($languageId, null, \PDO::PARAM_INT)
```

Comment:
I found more optimal solution, however less pretty :) 
Instead of loading Content Info, this can be done by DBMS, using SQL expression:
```sql
(SELECT CASE 
    WHEN initial_language_id = :languageId AND language_mask & :languageId <> 0 THEN :languageId | 1
    ELSE :languageId END 
 FROM ezcontentobject 
WHERE id = :contentId)
```
which could be written here verbatim as:
```php
'(SELECT CASE ' .
    'WHEN initial_language_id = :languageId AND language_mask & :languageId <> 0 THEN :languageId | 1 ' .
    'ELSE :languageId END ' .
'FROM ezcontentobject ' .
'WHERE id = :contentId)'
```

Later on `$q` needs the parameters used above, so before `$q->prepare()` call:
```php
$q->bindValue($language->id, ':languageId', \PDO::PARAM_INT);
$q->bindValue($contentId, ':contentId', \PDO::PARAM_INT);
```

This query could be written using our own expression wrapper (see below), however last time I did similar nested case query we didn't really like readability. 
POV ping @andrerom 

```php
$languageQuery = $this->dbHandler->createSelectQuery();
$languageQuery
    ->select(
        $languageQuery->expr->searchedCase(
            [
                $languageQuery->expr->lAnd(
                    $languageQuery->expr->eq(
                        $this->dbHandler->quoteColumn('initial_language_id'),
                        ':languageId'
                    ),
                    // wrap bitwise check into another "neq" to provide cross-DBMS compatibility
                    $languageQuery->expr->neq(
                        $languageQuery->expr->bitAnd(
                            $this->dbHandler->quoteColumn('language_mask'),
                            ':languageId'
                        ),
                        0
                    )
                ),
                $languageQuery->expr->bitOr(
                    ':languageId', 1
                )
            ],
            ':languageId'
        )
    )
    ->from('ezcontentobject')
    ->where(
        $languageQuery->expr->eq(
            'id',
            ':contentId'
        )
    )
;
```